### PR TITLE
Initial commit for aws 3 jobs: ebs_private_snapshot, rds_enable_versi…

### PR DESCRIPTION
**Pull request by lytran2000**
…on_update, rds_remove_public_endpoint
added negative and positive test cases for remediation

### CODE REVIEW

File: remediation_worker/jobs/rds_remove_public_endpoint/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Private rds snapshot
```
Comment (kshrutik) [inline remediation_worker/jobs/rds_remove_public_endpoint/README.md:1]: Disable public access to RDS instances

File: remediation_worker/jobs/rds_remove_public_endpoint/rds_remove_public_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                "DBInstances")[0]["PubliclyAccessible"]
+
+        # If it is public, set to private
+        if is_public:
+            logging.info(
+                "RDS database %s is public, setting to private.",
+                instance_id
+                )
+            try:
+                client.modify_db_instance(
```
Comment (kshrutik) [inline remediation_worker/jobs/rds_remove_public_endpoint/rds_remove_public_endpoint.py:90]: Add logging messages.

File: remediation_worker/jobs/rds_remove_public_endpoint/rds_remove_public_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+        Args:
+            instance_id ([str]): The ID of the RDS instance.
+            client: Instance of the AWS boto3 client.
+        """
+
+        logging.info("Beginning RDS public endpoint evaluation.")
+
+        # Check to see if the database
+        # is publicly accessible
```
Comment (kshrutik) [inline remediation_worker/jobs/rds_remove_public_endpoint/rds_remove_public_endpoint.py:78]: Add logging messages

File: remediation_worker/jobs/rds_remove_public_endpoint/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+This job disables rds public accessible to ensure it is not connected by any internet hosts.
```
Comment (kshrutik) [inline remediation_worker/jobs/rds_remove_public_endpoint/README.md:3]: This job removes public access from RDS instances and makes the instances private.

**Review by kshrutik — CHANGES_REQUESTED** [review]

File: remediation_worker/jobs/ebs_private_snapshot/ebs_private_snapshot.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        """Set snapshots to private.
+        :param client: Instance of the AWS boto3 client.
+        :param snapshot_id: The id of the EBS snapshot.
+        :param region: The region of the EBS snapshot.
+        :type snapshot_id: str.
+        :returns: Bool signaling success or failure
+        :rtype: bool
+        :raises: botocore.exceptions.ClientError
+        """
+        logging.info(f"Trying to remove Public access from {snapshot_id}")
```
[the remark this answers is not in the document]
Comment (lytran2000) [reply remediation_worker/jobs/ebs_private_snapshot/ebs_private_snapshot.py:69]: added code to catch exception and added a new testcase to raise ClientError exception.  Code ran cleanly if ebs is no longer existed
ltran-a02:jobs ltran$ python3 ebs_private_snapshot/ebs_private_snapshot.py "`cat finding/finding_snapshot.json`"
INFO:root:ebs_private_snapshot.py called - running now
INFO:root:  region: us-west-2
INFO:botocore.credentials:Found credentials in shared credentials file: ~/.aws/credentials
INFO:root:Removing Public access by executing client.describe_snapshot_attribute
INFO:root:Attribute = createVolumePermission
INFO:root:SnapshotId=snap-01a972f6209ba8d24
ERROR:root:Got Exception=InvalidSnapshot.NotFound

File: remediation_worker/jobs/rds_enable_version_update/rds_enable_version_update.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        :param database_id: The id of the RDS instanace.
+        :returns: updates flag for RDS auto upgrade minor version
+        """
+
+        logging.info(
+            "updating auto minor version upgrade for %s", database_id)
+
+        # convert the RDS auto minor version flag to True
+        try:
+            client.modify_db_instance(
```
[the remark this answers is not in the document]
Comment (lytran2000) [reply remediation_worker/jobs/rds_enable_version_update/rds_enable_version_update.py:67]: add log msgs:
logging.info("Setting AutoMinorVersionUpgrade to True")
logging.info(f"DBInstanceIdentifier={database_id}")

File: remediation_worker/jobs/ebs_private_snapshot/ebs_private_snapshot.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        
+        # Get the permissions of the snapshot, no exeption expected
```
[the remark this answers is not in the document]
Comment (lytran2000) [reply remediation_worker/jobs/ebs_private_snapshot/ebs_private_snapshot.py:71]: added log msgs as suggested
logging.info("Removing Public access by executing client.describe_snapshot_attribute")
logging.info("Attribute = createVolumePermission")
logging.info(f"SnapshotId={snapshot_id}")

File: remediation_worker/jobs/ebs_private_snapshot/minimum_policy.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "ModifyPrivateSnapShot",
+            "Effect": "Allow",
+            "Action": [
+                "DescribeSnapshotAttribute",
```
[the remark this answers is not in the document]
Comment (lytran2000) [reply remediation_worker/jobs/ebs_private_snapshot/minimum_policy.json:8]: updated "ec2:DescribeSnapshotAttribute",

File: remediation_worker/jobs/rds_remove_public_endpoint/README.md (hunk 1 above)
Comment (lytran2000) [reply remediation_worker/jobs/rds_remove_public_endpoint/README.md:1]: the description is below
This job disables public access to RDS instances to to ensure it is not connected by any internet hosts.

File: remediation_worker/jobs/rds_remove_public_endpoint/rds_remove_public_endpoint.py (hunk 3 above)
Comment (lytran2000) [reply remediation_worker/jobs/rds_remove_public_endpoint/rds_remove_public_endpoint.py:78]: logging.info("Beginning RDS public endpoint evaluation.")

        # Check to see if the database
        # is publicly accessible
        logging.info("executing client.describe_db_instances")
        logging.info(f"RDS instance={instance_id}")

File: remediation_worker/jobs/rds_remove_public_endpoint/README.md (hunk 1 above)
Comment (kshrutik) [reply remediation_worker/jobs/rds_remove_public_endpoint/README.md:1]: I'm asking you to change the job title from Private rds snapshot to Disable public access to RDS DBInstances.

File: remediation_worker/jobs/ebs_private_snapshot/README.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
+# Disable public access to RDS DBInstances.
```
Comment (kshrutik) [inline remediation_worker/jobs/ebs_private_snapshot/README.md:1]: Configure the EBS volume snapshot as private.

**Review by kshrutik — APPROVED** [review]

**Review by nandeshguru — APPROVED** [review]

**Review by nandeshguru — APPROVED** [review]

**Pull request merged**